File: URI/JavaScript/uri-1010.js
// URI - 1010

const readline = require('readline')

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
})

class Piece {
    constructor(id, amount, unitaryValue) {
        this.id = id,
        this.amount = amount,
        this.unitaryValue = unitaryValue
    }

    totalPrice() {
        return this.amount * this.unitaryValue
    }
}

rl.on('line', input => {
    let partData = input.split(' ')
    const piece1 = new Piece(parseInt(partData[0]), parseInt(partData[1]), parseFloat(partData[2]))

    rl.on('line', input => {
        partData = input.split(' ')
        const piece2 = new Piece(parseInt(partData[0]), parseInt(partData[1]), parseFloat(partData[2]))

        console.log('VALOR A PAGAR: R$ ' + (piece1.totalPrice + piece2.totalPrice).toFixed(2))

        rl.close()
    })
})

File: URI/Ruby/uri-1010.rb
#URI - 1010: https://www.urionlinejudge.com.br/judge/pt/problems/view/1010

class Piece
    def initialize(code, number_of_pieces, unitary_value)
        @code = code
        @number_of_pieces = number_of_pieces
        @unitary_value = unitary_value
    end
    def calculate_price
        price = @number_of_pieces * @unitary_value
        return price
    end
end

input = gets.split(' ')
Piece_A = Piece.new(input[0].to_i, input[1].to_i, input[2].to_f)
input = gets.split(' ')
Piece_B = Piece.new(input[0].to_i, input[1].to_i, input[2].to_f)
Total_payable = Piece_A.calculate_price + Piece_B.calculate_price

puts "VALOR A PAGAR: R$ #{'%.2f' % Total_payable}"